File: uv_10360.java
import java.util.*;
class box
{
    int x;
    int y;
    box(int x,int y){
    	this.x=x;
    	this.y=y;
    }

}

class Main{
	public static void main(String[] args){
		Scanner in = new Scanner(System.in);
		
		int t=in.nextInt();
		while(t-->0){
			int[][] a=new int[1025][1025];
		int[][] killed =new int[1025][1025];
			int d= in.nextInt();
			int n = in.nextInt();
			box maxC=null;
			int maxi=-1;
			
			while(n-->0){
				int r=in.nextInt();
				int col= in.nextInt();
				int p = in.nextInt();
				a[r][col]=p;
			for(int j= r-d;j<=r+d && j>=0 && j<1025;j++ ){
					for(int k= col-d;k<=col+d && k>=0 && k<1025;k++ ){
						killed[j][k]+=a[r][col];
					    if(killed[j][k]>maxi){
					    	maxi =killed[j][k];
					    	maxC=new box(j,k);
					    }
					    else if(killed[j][k]==maxi){
					    	if(j<maxC.x && k<maxC.y){
					    		maxC=new box(j,k);
					    		maxi=killed[j][k];
					    	}
					    }}
				
				}
			}	
	
        System.out.println(maxC.x+" "+maxC.y+" "+maxi);

		
		}
	}}

File: uv_10360.cpp
#include<iostream>
#include<vector>
using namespace std;


class coord{
	public:
    int x;
	int y;
    coord(){}
	 coord(int X,int Y){
		x=X;
		y=Y;
	}
	
};

int main()
{   
	int t;
    
    scanf("%d",&t);
    while(t--){
        int n,d,maxi=-1;
        vector<vector<int> > killed( 1025 , vector<int>(1025,0));
        vector<vector<int> > a( 1025 , vector<int>(1025,0));
	coord maxC(-1,-1);
        scanf("%d",&d);
        scanf("%d",&n);
          while(n--){
            int x,y,p;
            cin>>x>>y>>p;
            a[x][y]=p;
            for(int i=x-d ;i<=x+d && i<1025 && i>=0;i++){
                 for(int j=y-d ;j<=y+d && j<1025 && j>=0;j++){
                       killed[i][j]+=a[x][y];
                       if(killed[i][j]>maxi){
                          maxi=killed[i][j];
                          maxC.x=i;
                          maxC.y=j;
        }
          else if(killed[i][j]==maxi){
              if(maxC.x>i && maxC.y>j){
                 maxi= killed[i][j];
                 maxC.x=i;
                 maxC.y=j;
            }
        }
}

        }}
        
       printf("%d %d %d\n",maxC.x,maxC.y,maxi); 
     }
    return 0;
}
